charts/federation-v2/charts/controllermanager/templates/federationconfig.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederationConfig
+metadata:
+  name: federation-v2
+  namespace: {{ .Release.Namespace }}
+spec:
+  limited-scope: {{ .Values.global.limitedScope }}
+{{- if .Values.registryNamespace }}
+  registry-namespace: {{ .Values.registryNamespace }}
+{{- else }}
+{{- if .Values.global.limitedScope }}
```

Comment:
What do you think about a new flag (e.g. `registryInFederationNamespace: true|false`) to avoid having to hard-code a registry namespace in the `FederationConfig` in the case of a limited-scope deployment without `registryNamespace` being specified?

Reply:
As we discussed before, `registry-namespace` should be always support for potential `cluster` registry sharing case. 

Therefore, we should make `registry-namespace` configurable. It is the reason to put the variable into `FederationConfig`. 
And in such case, we still need a default namespace if it is cluster federation and with no `registry-namespace` specified. So, `kube-multicluster-public` is hardcoded here.

I think it is clear in feature. 
I do not understand why flag `registryInFederationNamespace` helps in this case:   
    If it is true, we've already handled it with `registry-namespace: {{ .Release.Namespace}}` 
    If it is false, we still need a `resigtry-namespace`.

Reply:
I'm not sure how I gave you the impression that I wanted to avoid having `registry-namespace` be configurable.  That was not my intent.

In all cases, the federation namespace is the namespace in which the controller manager is deployed.  If the registry namespace is not provided by the user, it is defaulted to `kube-multicluster-public` if cluster-scoped and to the federation namespace if namespace-scoped.  

I think is strange to supply the federation namespace to the controller manager via the downward api, but supply the same value for the registry namespace via a hard-coded value.  My suggestion for `registryInFederationNamespace` would be to indicate that the controller manager should use the federation namespace, and it would only be respected if `registry-namespace` was empty.  The helm chart would provide this configuration for a limited-scope deployment where the user did not specify `registry-namespace`.  This would ensure the discoverability of the registry namespace without having to hard-code it.

Reply:
Ah, nevermind.  I'm splitting hairs.